py/server/deephaven/experimental/iceberg.py

```diff
@@ -140,6 +144,222 @@ def j_object(self) -> jpy.JType:
+                before switching to non-dictionary encoding, never evaluated for non-String columns, defaults to
+                2^20 (1,048,576)
+            target_page_size (Optional[int]): the target page size in bytes, if not specified, defaults to
+                2^20 bytes (1 MiB)
+            snapshot_id (Optional[int]): The identifier of the snapshot to load for updating.; if omitted, the most
+                recent snapshot will be used.
+            table_definition (Optional[TableDefinitionLike]): the TableDefinition to use when writing Iceberg data
+                files, instead of the one implied by the table being written itself. This definition can be used to skip
+                some columns or add additional columns with null values.
+                When passing this value to an IcebergTableWriter, this table definition should either:
+                - Not be provided, in which case the definition will be derived from the writer instance, or
+                - Match the writer's table definition if it is provided.
+
+        Raises:
+            DHError: If unable to build the instructions object.
+        """
+
+        try:
+            builder = self.j_object_type.builder()
+
+            if isinstance(tables, Table):
+                builder.addTables(tables.j_table)
+            elif isinstance(tables, Sequence):
+                for table in tables:
+                    builder.addTables(table.j_table)
+
+            if partition_paths:
+                if isinstance(partition_paths, str):
+                    builder.addPartitionPaths(partition_paths)
+                elif isinstance(partition_paths, Sequence):
+                    for partition_path in partition_paths:
+                        builder.addPartitionPaths(partition_path)
+
+            if compression_codec_name:
+                builder.compressionCodecName(compression_codec_name)
+
+            if maximum_dictionary_keys:
+                builder.maximumDictionaryKeys(maximum_dictionary_keys)
+
+            if maximum_dictionary_size:
+                builder.maximumDictionarySize(maximum_dictionary_size)
+
+            if target_page_size:
+                builder.targetPageSize(target_page_size)
+
+            if snapshot_id:
+                builder.snapshotId(snapshot_id)
+
+            if table_definition:
+                builder.tableDefinition(TableDefinition(table_definition).j_table_definition)
+
+            if data_instructions:
+                builder.dataInstructions(data_instructions.j_object)
+
+            self._j_object = builder.build()
+
+        except Exception as e:
+            raise DHError(e, "Failed to build Iceberg write instructions") from e
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self._j_object
+
+
+class SchemaSpec(JObjectWrapper):
+    """
+    A specification for extracting the schema from a table.
+    """
```

Comment:
a bit confused about the name and description.

Reply:
Renamed it to SchemaProvider, let me know if this is more clear.